org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/TelemetryManager.java

```diff
@@ -152,6 +163,9 @@ public void onBuildFinished(long buildFinishedTime) {
 		buildFinishedElapsedTime = buildFinishedTime - this.languageServerStartTime;
 
 		properties.add("buildToolNames", buildToolNamesList);
+		if (hasKotlinGradleFiles) {
+			properties.addProperty("hasKotlinGradleFiles", Boolean.toString(hasKotlinGradleFiles));
```

Comment:
I'm not fond of build tool specific code. I'd rather prefer we stay build tool agnostic as much as possible.
I'd rather prefer something like buildFile=pom.xml|build.gradle|build.gradle.kts|... if possible

Reply:
it could be done, but I'm not sure what relevant data we'd get for non-Gradle projects.

- All Maven projects would be "pom.xml" at the top-level. I could report it just for the sake of doing so
-  All Eclipse projects could be "build.properties", or just "none". Is it worth reporting whether an Eclipse project has a top-level pom file even though the MavenProjectImporter (runs with priority over EclipseProjectImporter) failed to detect it as a Maven project ?
- For invisible projects there's "none"

Gradle is the only really interesting options as it permits 2 different styles of build.

Reply:
@fbricon Ok, this should be fixed now. `buildFileNames` only reports for gradle now and we can report a list in case we need to expand to multiple types of build files in the future.

Reply:
I'll merge this for now, and in the future if we really want to know more about other build system files, we could add to IBuildSupport. Frameworks as opposed to just files would be interesting to detect.